tdrs-backend/tdpservice/data_files/admin.py

```diff
@@ -60,7 +60,44 @@ def data_file_summary(self, obj):
         return format_html("<a href='{url}'>{field}</a>",
                            field=f'{df.id}' + ":" + df.get_status(),
                            url=f"{DOMAIN}/admin/parsers/datafilesummary/{df.id}/change/")
-
+    
+    class by_submission_date(admin.SimpleListFilter):
+        """filter data files by month."""
+        title = 'Submission Date'
+        parameter_name = 'Submission Day/Month/Year'
+
+        def lookups(self, request, model_admin):
+            """Return a list of tuples."""
+            return [
+                ('1', 'Yesterday'),
+                ('0', 'Today'),
+                ('7', 'Past 7 days'),
+                ('30', 'This month'),
+                ('365', 'This year'),
+            ]
+        
+        def queryset(self, request, queryset):
+            """Return a queryset."""
+            from datetime import datetime, timedelta, timezone
+            yesterday = (datetime.now(tz=timezone.utc) - timedelta(days=1)).replace(hour=0, minute=0, second=0, microsecond=0)
+            this_month = datetime.now(tz=timezone.utc).replace(day=1)
+            this_year = datetime.now(tz=timezone.utc).replace(month=1, day=1)
+            if self.value() == '1':
+                query_set_ids = [df.id for df in queryset if df.created_at.replace(hour=0, minute=0, second=0, microsecond=0) == yesterday]
+                return queryset.filter(id__in=query_set_ids)
+            elif self.value() in ['0', '7']:
+                last_login__lte=datetime.now(tz=timezone.utc) - timedelta(days=int(self.value()))
+                query_set_ids = [df.id for df in queryset if df.created_at >= last_login__lte]
+                return queryset.filter(id__in=query_set_ids)
+            elif self.value() == '30':
+                query_set_ids = [df.id for df in queryset if df.created_at >= this_month]
+                return queryset.filter(id__in=query_set_ids)
+            elif self.value() == '365':
+                query_set_ids = [df.id for df in queryset if df.created_at >= this_year]
+                return queryset.filter(id__in=query_set_ids)
```

Comment:
Can the logic for all these blocks starting at line 88 through 95 be condensed into a single `elif` and just use a `timedelta`?

Reply:
Unfortunately no! The reason is definition of last month is not 30days delta, but depending on where in that month we are can be  31~59 days

Reply:
Oh I see now, thank you for clarifying!